preprocessor/src/main/java/com/azure/autorest/preprocessor/tranformer/Transformer.java

```diff
@@ -374,4 +379,51 @@ private Parameter createContentLengthParameter(Operation operation, Parameter bo
     contentType.getLanguage().setJava(language);
     return contentType;
   }
+
+  private static void deduplicateParameterNames(List<Parameter> parameters) {
+    Set<String> parameterNames = new HashSet<>();
+    ListIterator<Parameter> iter = parameters.listIterator();
+    while (iter.hasNext()) {
+      Parameter parameter = iter.next();
+      if (parameterNames.contains(parameter.getLanguage().getJava().getName())) {
+        // use a new Parameter, in case the original one is referenced by multiple requests
+        Parameter newParameter = new Parameter();
+        shallowCopy(parameter, newParameter, Parameter.class);
+        newParameter.setLanguage(new Languages());
+        shallowCopy(parameter.getLanguage(), newParameter.getLanguage(), Languages.class);
+        newParameter.getLanguage().setJava(new Language());
+        shallowCopy(parameter.getLanguage().getJava(), newParameter.getLanguage().getJava(), Language.class);
+        newParameter.getLanguage().getJava().setName(parameter.getLanguage().getJava().getName() + "Param");
+
+        iter.set(newParameter);
+
+        parameter = newParameter;
+      }
+
+      parameterNames.add(parameter.getLanguage().getJava().getName());
+    }
+  }
```

Comment:
Not sure if this is the best solution. @jianghaolu

Reply:
For now, reverted to modify the java name, instead of making a new `Parameter` object.

Reply:
Fixing param names should be done in pre-processor - so I think this is the right place for it.

And yes I don't think making a copy is necessary.